### PR TITLE
Made it work properly (at all) on py3 again.

### PR DESCRIPTION
**Pull request by rostob**
It should also still work on py2.

### CODE REVIEW

File: SpiffyTitles/plugin.py [hunk 1]
```diff
@@ -129,9 +127,14 @@ def handler_dailymotion(self, url, info, channel):
...
-                            Template(self.registryValue("dailymotionVideoTitleTemplate",
-                                                        channel=channel).decode("utf-8"))
+                        if sys.version_info[0] >= 3:
+                            dailymotion_template = Template(
+                                self.registryValue("dailymotionVideoTitleTemplate",
+                                                   channel=channel))
+                        else:
+                            dailymotion_template = Template(
+                                self.registryValue("dailymotionVideoTitleTemplate",
+                                                   channel=channel).decode("utf-8"))
```
Comment (butterscotchstallion) [inline SpiffyTitles/plugin.py]: Maybe we could write a method for this. This bit of code is repeated for each template type, so I think it's worth refactoring a bit. Consider the following bit of psuedocode:

`
def get_template (self, **kwargs):
  is_py3 = sys.version_info[3] >= 3
  # return template based on channel, name, and whether we're using py3 here
  # and conditionally apply .decode("utf-8")
`

**Comment by butterscotchstallion** [comment]
Also, @rostob please try the latest master code. I have accepted several PRs addressing Python 2/3 compatibility

**Comment by lunchdump** [comment]
The latest version works for me on both Python 2 and Python 3

**Comment by rostob** [comment]
@lunchdump The latest master does not work on Python 3. It loads, but it does not work.
It does not give any hints as to why it does not work either, that is why I needed to fix it.
If my pull request gets accepted, I would appreciate feedback on if it still functions properly with Python 2.

File: SpiffyTitles/plugin.py (hunk 1 above)
Comment (rostob) [inline SpiffyTitles/plugin.py]: I agree, and I have refactored it. I will push my commits now.

**Comment by lunchdump** [comment]
I have 2 bots running both py2 and py3, both are using this plugin without local changes, both work fine.

Could you provide more information about your environment, or steps to reproduce? Are you sure you have all the dependencies in requirements.txt installed? Have you enabled debug logging? (!config supybot.log.level DEBUG)

**Comment by rostob** [comment]
In the current master, only the default handler works on Python3.
If you have the default handler disabled, no titles are handled, which is my environment.
To reproduce, disable the default handler and try any URL with a handler, or just look at the code:
`grep -B1 'utf-8' plugin.py`

These `.decode("utf-8")` makes it not work, silently, on Python3.

On Python2, in the current master, the default handler can probably not handle unicode in its template.
To test, add some unicode to the default template string, when running with Python2.

My commits fixes all these issues.
I also noticed that the "t" command is broken.
I have now also fixed that.

File: SpiffyTitles/plugin.py [hunk 2]
```diff
@@ -430,17 +430,43 @@ def t(self, irc, msg, args, query):
...
-            if url:
-                title = self.get_title_by_url(query, channel)
-        except:
-            pass
-
-        if title is not None and title:
-            irc.queueMsg(ircmsgs.privmsg(channel, title))
+            if not handled:
+                if title is not None and title:
+                    if "#" in channel:
```
Comment (butterscotchstallion) [inline SpiffyTitles/plugin.py]: Limnoria has a way to check if something is a channel without doing something like this. There is something to check if it's a PM too. Check my other repos if you can't find it.

**Pull request merged**